addons/l10n_ng/models/template_ng.py

```diff
@@ -0,0 +1,53 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+from odoo.addons.account.models.chart_template import template
+
+
+class AccountChartTemplate(models.AbstractModel):
+    _inherit = 'account.chart.template'
+
+    def _get_ng_additional_accounts(self):
+        return {
+            'withholding': {
+                'name': _("Withholding Tax on Purchases"),
+                'code': '252001',
+                'account_type': 'liability_current',
+                'reconcile': False,
+            },
+            'withholding_transitional': {
+                'name': _("Withholding Tax on Purchases - Transition Account"),
+                'code': '252002',
+                'account_type': 'liability_current',
+                'reconcile': False,
+            },
+        }
+
+    @template(model='account.account')
+    def _get_account_account(self, template_code):
+        # Nigerian companies are fine with using the generic COA
+        # but we need to add Nigeria-specific taxes and a tax report
+        # so we need to parse the generic coa csv and call
+        # generic coa template functions here to make it work together
+        if template_code == 'ng':
+            res = self._parse_csv('generic_coa', 'account.account')
+            res.update(self._get_ng_additional_accounts())
+            return {f'l10n_ng_{k}': v for k, v in res.items()}
+        return super()._get_account_account(template_code)
+
+    @template('ng')
+    def _get_ng_template_data(self):
+        # this is done this way in case something changes in the generic coa
+        # so the changes will be reflected here as well
+        res = self._get_generic_coa_template_data()
+        return {k: f'l10n_ng_{v}' for k, v in res.items() if k not in ('name', 'country')}
```

Comment:
It's not actually doing much technically, but I'd like to be explicit and respect the interface.

```suggestion
        return {
            'name': "Nigerian Chart of Accounts",
            'country': res['country'],
            **{k: f'l10n_ng_{v}' for k, v in res.items() if k not in ('name', 'country')}
        }
```